sshd-core/src/main/java/org/apache/sshd/common/config/keys/KeyUtils.java

```diff
@@ -144,20 +161,52 @@ public static PosixFilePermission validateStrictKeyFilePermissions(Path path, Li
         }
 
         if (perms.contains(PosixFilePermission.OTHERS_EXECUTE)) {
-            return PosixFilePermission.OTHERS_EXECUTE;
+            PosixFilePermission p = PosixFilePermission.OTHERS_EXECUTE;
+            return new Pair(String.format("Permissions violation (%s)", p), p);
         }
 
         if (OsUtils.isUNIX()) {
             PosixFilePermission p = IoUtils.validateExcludedPermissions(perms, STRICTLY_PROHIBITED_FILE_PERMISSION);
             if (p != null) {
-                return p;
+                return new Pair(String.format("Permissions violation (%s)", p), p);
             }
 
             if (Files.isRegularFile(path, options)) {
                 Path parent = path.getParent();
                 p = IoUtils.validateExcludedPermissions(IoUtils.getPermissions(parent, options), STRICTLY_PROHIBITED_FILE_PERMISSION);
                 if (p != null) {
-                    return p;
+                    return new Pair(String.format("Parent permissions violation (%s)", p), p);
+                }
+            }
+        }
+
+        String current = OsUtils.getCurrentUser();
+        String owner = IoUtils.getFileOwner(path, options);
+
+        if (GenericUtils.isEmpty(owner)) {
+            // we cannot get owner
+            // general issue: jvm does not support permissions
+            // security issue: specific filesystem does not support permissions
+            return null;
+        }
+
+        Set<String> expected = new HashSet<>();
+        expected.add(current);
+        if (OsUtils.isUNIX()) {
+            // Windows Administrator was considered however
+            // in Windows most liklely a group is in used.
+            expected.add(OsUtils.ROOT_USER);
+        }
+
+        if (!expected.contains(owner)) {
+            return new Pair(String.format("Owner violation (%s)", owner), owner);
+        }
+
+        if (OsUtils.isUNIX()) {
+            if (Files.isRegularFile(path, options)) {
+                String parentOwner = IoUtils.getFileOwner(path.getParent(), options);
+                if (!expected.contains(parentOwner)) {
+                    return new Pair(String.format("Parent owner violation (%s)", parentOwner), parentOwner);
                 }
```

Comment:
there is no need, if we do have owner file system support, then it must exist, if return null then it won't be in the collection and we fail.